libgalois/include/katana/Mem.h

```diff
@@ -39,6 +39,21 @@ void Prealloc(size_t pagesPerThread, size_t bytes);
 KATANA_EXPORT
 void Prealloc(size_t pages);
 
+/**
+ * Ensures an amount of memory be free on each thread. The allocation size is
+ * given in pages per thread and total bytes which will be divided amongst the
+ * threads.
+ */
+KATANA_EXPORT
+void EnsureFree(size_t pagesPerThread, size_t bytes);
```

Comment:
I don't know how I feel about the name `EnsureFree`. Would `EnsureUnallocated` make sense (or even "ensure prealloc") ?

Reply:
`EnsureMemoryAvailable`?

Reply:
Sounds good.

Reply:
`EnsurePreallocated` is better I think. Does that work?